teps/0079-tekton-catalog-support-tiers.md

```diff
@@ -100,14 +128,460 @@ requirements, and provenance attestations can be made to meet software supply ch
+    revision: main
+    
+  # official catalog  
+  - name: tekton-official
+    org: tektoncd
+    type: official
+    provider: github
+    url: https://github.com/tektoncd/catalog-official
+    revision: main
+```
+
+This provides extensibility to the support tiers: we can add additional
+Catalogs for other support tiers as needed in the future. 
+
+#### Community and Official Catalogs in the Tekton CLI
+
+Today, users can use the CLI to install resources from Catalogs by
+passing in the Catalog name to the `--from` argument as shown in the
+examples below. This will add a label to the resource indicating the
+source Catalog: `hub.tekton.dev/catalog=<catalog-name>`.
+
+```shell
+# Community Catalog
+
+$ tkn hub install task golang-lint --version 0.3 --from tekton-community
+
+Task golang-lint(0.3) installed in default namespace
+
+$ kubectl describe task.tekton.dev/golang-lint
+
+Name:         golang-lint
+Namespace:    default
+Labels:       app.kubernetes.io/version=0.3
+              hub.tekton.dev/catalog=tekton-community
+...
+ 
+# Official Catalog
+
+$ tkn hub install task golang-fuzz --version 0.1 --from tekton-official
+
+Task golang-fuzz(0.1) installed in default namespace
+
+$ kubectl describe task.tekton.dev/golang-fuzz
+
+Name:         golang-fuzz
+Namespace:    default
+Labels:       app.kubernetes.io/version=0.1
+              hub.tekton.dev/catalog=tekton-official
+...
+```
+
+#### Community and Official Catalogs in the Cluster
+
+When resources are installed in a cluster, without using the CLI, 
+it may be difficult to identify which Tekton Catalog it came from
+because they won't have the labels added by the CLI.
+
+To make it easy for users to identify the source Catalog, we propose 
+adding a `tekton.dev/catalog` annotation to the resource yaml file, 
+as such:
+
+```yaml
+# resource from the community catalog
+annotations:
+  tekton.dev/pipelines.minVersion: "0.17.0"
+  tekton.dev/tags: golang-lint
+  tekton.dev/catalog: community
+
+# resource from the official catalog
+annotations:
+  tekton.dev/pipelines.minVersion: "0.30.0"
+  tekton.dev/tags: golang-fuzz
+  tekton.dev/catalog: official
+```
+
+#### Community and Official Catalogs in Tekton Bundles
+
+We propose publishing the resources in Catalogs as bundles separate OCI registries:
+* Community Catalog - `gcr.io/tekton-releases/catalog/upstream/<resource-name>:<resource-version>`
+* Official Catalog - `gcr.io/tekton-releases/catalog-official/upstream/<resource-name>:<resource-version>`
+
+Users can use the applicable reference to fetch and use the resource, as such:
+
+```yaml
+# bundle from a resource in the community catalog
+taskRef:
+    name: golang-lint
+    bundle: gcr.io/tekton-releases/catalog/upstream/golang-lint:0.1
+
+# bundle from a resource in the official catalog
+taskRef:
+  name: golang-fuzz
+  bundle: gcr.io/tekton-releases/catalog-official/upstream/golang-fuzz:0.1
+```
+
+#### Community and Official Catalogs in Remote Resolution
+
```

Comment:
another thing is I wonder if between remote resolution and catalogs we can validate the origin of the tasks not being tampered by way of a encrypted signature.. maybe another topic for another day for the tekton chains folks!